pkg/util/containers/metrics/system/filter_container.go

```diff
@@ -0,0 +1,172 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+
+package system
+
+import (
+	"sync"
+
+	"github.com/DataDog/datadog-agent/comp/core/workloadmeta"
+	"github.com/DataDog/datadog-agent/pkg/util/cgroups"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type trieNode struct {
+	children map[rune]*trieNode
+	cid      string
+}
+
+func newTrieNode() *trieNode {
+	return &trieNode{
+		children: make(map[rune]*trieNode),
+	}
+}
+
+type suffixTrie struct {
+	root *trieNode
+}
+
+func newSuffixTrie() *suffixTrie {
+	return &suffixTrie{
+		root: newTrieNode(),
+	}
+}
+
+func reverse(s string) string {
```

Comment:
By reversing, you target matching optimization by expecting path end is more unique than path start (which is true), however it means you basically don't benefit from memory savings of the trie as you're going to store all paths starts as many times as entries (mostly).